web/pgadmin/browser/static/js/browser.js

```diff
@@ -1699,6 +1699,8 @@ define('pgadmin.browser', [
                 // If the current node's inode is false
                 if (__ctx.i && !__ctx.t.isInode(__ctx.i)) {
                   __ctx.t.setInode(__ctx.i, {success: _appendNode});
+                  // Open the collection node.
```

Comment:
Shouldn't the node be re-selected if it is renamed? If anything is removed then collection should be selected.

Reply:
If the collection node has a single child node that child node will get auto-selected on update/rename and if the user removes the child node it will select the collection node if no other child node is present.

Reply:
OK.